service/controller/resource/clusterrolebinding/resource.go

```diff
@@ -18,26 +18,35 @@ const (
 type Config struct {
 	K8sClient kubernetes.Interface
 	Logger    micrologger.Logger
+
+	PodSecurityPolicyName string
 }
 
 // Resource implements the config map resource.
 type Resource struct {
 	k8sClient kubernetes.Interface
 	logger    micrologger.Logger
+
+	podSecurityPolicyName string
 }
 
 // New creates a new configured config map resource.
 func New(config Config) (*Resource, error) {
 	if config.K8sClient == nil {
-		return nil, microerror.Maskf(invalidConfigError, "config.K8sClient must not be empty")
+		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", config)
 	}
 	if config.Logger == nil {
-		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	if config.PodSecurityPolicyName == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.PodSecurityPolicyName must not be empty", config)
 	}
 
 	newService := &Resource{
 		k8sClient: config.K8sClient,
 		logger:    config.Logger,
+
+		podSecurityPolicyName: config.PodSecurityPolicyName,
```

Comment:
Did you test upgrades? Is the PSP name changed during the upgrade and everything is fine?

Reply:
Good point. I will investigate.

Reply:
The previous hardcoded PSP is `kvm-operator-psp` which should still be created by the `thiccc` operator. In the case of a flattened operator, the required PSP is created when the operator is deployed via helm so it must exist by the time a cluster is being upgraded as far as I can tell. I will deploy this branch to make sure that my thinking is correct here.

Reply:
> the required PSP is created when the operator is deployed via helm

If this is in `resource` package then this code is executed only during CR reconciliation. So surely not right after deployment. I'm not clear on what the PSP is used for though.

Reply:
I don't fully understand `kvm-operator` myself, but I believe this PSP is used for the pods on the control plane that run VMs that run the TC nodes. So basically the operator pod and the node pods share the same PSP.

Reply:
Good call on testing upgrades. Sussed out a few problems in the process.